blockprotocol/blockprotocol
# Allow creating new entity types with properties/links

## 🌟 What is the purpose of this PR?

This copies the approach in HASH allowing you to pass a base64 encoded JSON schema as the `draft` query param to the entity type editor in order to edit a "draft" entity type, i.e, to open an entity type which doesn't exist but with title/description set. This then allows you to create the entity type with links/properties




## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203623179643287/1204060043683838/f) _(internal)_

## 🚫 Blocked by




N/A

## 🔍 What does this change?




- No longer creating an entity type in the new entity type modal, instead checking for uniqueness and redirecting to the entity type editor with `?draft=….` set
- Don't fetch an entity type if `?draft=` query param set, but parse it and preset UI info with it
- Edit UI to indicate draft state
- When submitting form in draft state, create entity type and redirect instead of updating entity type

## 📜 Does this require a change to the docs?



N/A

## ⚠️ Known issues



No validation of draft query param. See https://hashintel.slack.com/archives/C02LG39FJAU/p1677780147133839

## 🐾 Next steps



Added validation

## 🛡 What tests cover this?



N/A

## ❓ How to test this?



1. Create entity type
2. See draft state of editor
3. Add properties/links
4. Click create
5. Refresh and see values persisted on version 1
